### PR TITLE
LPS-184111 Check if it exists before adding to avoid duplicate key.

### PR DESCRIPTION
**Pull request by manuelRives**
**7.3 to 7.4 com.liferay.dynamic.data.mapping.service upgrade results in a Index duplicate entry**

The issue is the same as [LPS-144100](https://issues.liferay.com/browse/LPS-144100) but in the other add method.

The issue appears as a duplicate entry:

Caused by: com.liferay.portal.kernel.log.LogSanitizerException: java.sql.BatchUpdateException: ORA-00001: unique constraint (LIFERAYPROMG.IX_998C30ED) violated_ [Sanitized]

### CODE REVIEW

**Comment by manuelRives** [comment]
ci:test:sf

**Comment by manuelRives** [comment]
ci:test:stable
ci:test:relevant
ci:test:forms

**Comment by manuelRives** [comment]
Hi @liferay-forms I've reviewed all the failed tests and none of them seems to be related with my changes.

Regards.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java [hunk 1]
```diff
@@ -239,11 +239,19 @@ private void _addDataDefinitionFieldLinks(
...
+
+			if (fieldSetDDMStructureId != 0) {
+				if (deDataDefinitionFieldLink == null) {
+					_deDataDefinitionFieldLinkLocalService.
+						addDEDataDefinitionFieldLink(
+							groupId, classNameId, dataDefinitionId,
+							fieldSetDDMStructureId, ddmFormField.getName());
+				}
 
 				_addDataDefinitionFieldLinks(
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java:256]: @manuelRives can we simplify this code to:

```
long fieldSetDDMStructureId = GetterUtil.getLong(
	ddmFormField.getProperty("ddmStructureId"));

if (fieldSetDDMStructureId == 0) {
	continue;
}

DEDataDefinitionFieldLink deDataDefinitionFieldLink =
	_deDataDefinitionFieldLinkLocalService.
		fetchDEDataDefinitionFieldLinks(
			classNameId, dataDefinitionId, fieldSetDDMStructureId,
			ddmFormField.getName());

if (deDataDefinitionFieldLink == null) {
	_deDataDefinitionFieldLinkLocalService.
		addDEDataDefinitionFieldLink(
			groupId, classNameId, dataDefinitionId,
			fieldSetDDMStructureId, ddmFormField.getName());
}

_addDataDefinitionFieldLinks(
	classNameId, dataDefinitionId,
	ddmFormField.getNestedDDMFormFields(), groupId);
```
? 

This way we don't need to `fetchDEDataDefinitionFieldLinks` if `fieldSetDDMStructureId == 0`.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by manuelRives** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**